kvdb-rocksdb/src/lib.rs

```diff
@@ -596,6 +596,22 @@ impl Database {
 			.unwrap_or(0)
 	}
 
+	/// The number of keys in a column (estimated).
+	/// Does not take into account the data in the overlay.
+	pub fn num_keys(&self, col: u32) -> io::Result<u64> {
+		const ESTIMATE_NUM_KEYS: &str = "rocksdb.estimate-num-keys";
+		match *self.db.read() {
+			Some(ref cfs) => {
+				let cf = cfs.cf(col as usize);
+				match cfs.db.property_int_value_cf(cf, ESTIMATE_NUM_KEYS) {
+					Ok(estimate) => { Ok(estimate.unwrap_or_default())},
+					Err(err_string) => Err(other_io_err(err_string))
+				}
+			}
+			None => Err(other_io_err("Failed to take read lock on the db"))
```

Comment:
this is not failing to take a read lock, it can happen if we're restoring the database in other thread (e.g. by calling `restore` after warp-sync), I think in this case we should return 0, or we can return an `Option`